pkg/sql/importer/import_pgdump.go

```diff
@@ -0,0 +1,581 @@
+		// Otherwise, we silently ignore the drop statement and continue with the import.
+		for _, name := range names {
+			tableName := name.ToUnresolvedObjectName().String()
+			if err := p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+				descsCol := p.ExtendedEvalContext().Descs
+				err := descsCol.Direct().CheckObjectCollision(
+					ctx,
+					txn,
+					parentID,
+					keys.PublicSchemaID,
+					tree.NewUnqualifiedTableName(tree.Name(tableName)),
+				)
+				if err != nil {
+					return errors.Wrapf(err, `drop table "%s" and then retry the import`, tableName)
+				}
+				return nil
+			}); err != nil {
+				return err
+			}
+		}
+	case *tree.BeginTransaction, *tree.CommitTransaction:
+	case *tree.Insert, *tree.CopyFrom, *tree.Delete:
+		// handled during the data ingestion pass.
+	case *tree.CreateExtension, *tree.CommentOnDatabase, *tree.CommentOnTable,
+		*tree.CommentOnIndex, *tree.CommentOnConstraint, *tree.CommentOnColumn, *tree.SetVar, *tree.Analyze,
+		*tree.CommentOnSchema:
+		// These are the statements that can be parsed by CRDB but are not
+		// supported, or are not required to be processed, during an IMPORT.
+		// - ignore txns.
+		// - ignore SETs and DMLs.
+		// - ANALYZE is syntactic sugar for CreateStatistics. It can be ignored
+		// because the auto stats stuff will pick up the changes and run if needed.
+		return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T statement: %s", stmt, stmt))
+	case *tree.CreateType:
+		return errors.New("IMPORT PGDUMP does not support user defined types; please" +
+			" remove all CREATE TYPE statements and their usages from the dump file")
+	case *tree.Grant:
+		for i, db := range stmt.Targets.Databases {
+			if err := checkCatalogName(db, tree.Name(handler.dumpDatabaseName)); err != nil {
+				return err
+			}
+			stmt.Targets.Databases[i] = TempImportPGDumpDBName
+		}
+		for _, schema := range stmt.Targets.Schemas {
+			if schema.ExplicitCatalog {
+				if err := checkCatalogName(schema.CatalogName, tree.Name(handler.dumpDatabaseName)); err != nil {
+					return err
+				}
+				schema.CatalogName = TempImportPGDumpDBName
+			}
+		}
+		for _, table := range stmt.Targets.Tables.TablePatterns {
+			normalizedTable, err := table.NormalizeTablePattern()
+			if err != nil {
+				return err
+			}
+			tableName, ok := normalizedTable.(*tree.TableName)
+			if !ok {
+				return errors.New("cannot get the table name of grant target")
+			}
+			if tableName.ExplicitCatalog {
+				if err := checkCatalogName(tableName.CatalogName, tree.Name(handler.dumpDatabaseName)); err != nil {
+					return err
+				}
+				tableName.CatalogName = TempImportPGDumpDBName
+			}
+		}
+		handler.BufferedDDLStmts = append(handler.BufferedDDLStmts, formatStatement(stmt))
+	case error:
+		if !errors.Is(stmt, errCopyDone) {
+			return stmt
+		}
+	default:
+		return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T statement: %s", stmt, stmt))
```

Comment:
How complete is `bufferDDLPostgresStatement`? Does it cover all of the statements that may be found in a pgdump, and therefore this `default` case is to guard against something completely unexpected? Or is this just a subset?